Core/src/org/sleuthkit/autopsy/commonfilesearch/CommonAttributePanel.java

```diff
@@ -670,10 +675,102 @@ private void intraCaseRadioActionPerformed(java.awt.event.ActionEvent evt) {//GE
     }//GEN-LAST:event_intraCaseRadioActionPerformed
 
     private void searchButtonActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_searchButtonActionPerformed
-        search();
-        SwingUtilities.windowForComponent(this).dispose();
+        checkDataSourcesAndSearch();
+        this.dispose();
     }//GEN-LAST:event_searchButtonActionPerformed
 
+    /**
+     * If the settings reflect that a inter-case search is being performed,
+     * checks that the data sources in the current case have been processed with
+     * Correlation Engine enabled and exist in the central repository. Promting
+     * the user as to whether they still want to perform the search in the case
+     * any data sources are unprocessed. If the settings reflect that a
+     * intra-case search is being performed, it just performs the search.
+     *
+     * Notes: - Does not check that the data sources were processed into the
+     * current central repository instead of another. - Does not check that the
+     * appropriate modules to make all correlation types available were run. -
+     * Does not check if the correlation engine was run with any of the
+     * correlation properties properties disabled.
+     */
+    @Messages({"CommonAttributePanel.incompleteResults.introText=Results may be incomplete. Not all data sources in the current case were ingested into the current Central Repository. The following data sources have not been processed:",
+        "CommonAttributePanel.incompleteResults.continueText=\n\n Continue with search anyway?",
+        "CommonAttributePanel.incompleteResults.title=Search may be incomplete"
+    })
+    private void checkDataSourcesAndSearch() {
+        new SwingWorker<List<String>, Void>() {
+            //if the eamdb is enabled and an instance is able to be retrieved check if each data source has been processed into the cr 
+            @Override
+            protected List<String> doInBackground() throws Exception {
+                List<String> unCorrelatedDataSources = new ArrayList<>();
+                if (interCaseRadio.isSelected()) {
+                    if (EamDb.isEnabled() && EamDb.getInstance() != null) {
```

Comment:
You could reduce the nesting by checking the opposite of these two ifs and returning immediately, i.e., 
if (! radio.isSelected()) return xxx
if(! (that eam enabled thing) return xxx
(continue)

Reply:
Alright I generally try to avoid having multiple returns if I can, but I agree in this case it will improve readability quite a bit due to the depth of the nesting.